pkg/app/piped/platformprovider/kubernetes/loader.go

```diff
@@ -166,16 +167,50 @@ func (l *loader) LoadManifests(ctx context.Context) (manifests []Manifest, err e
 		err = fmt.Errorf("unsupport templating method %v", l.templatingMethod)
 	}
 
+	// Refine the namespace when reading manifests from git repo
+	// because the namespace is determined not only by its own namespace on the file but also the namespace on the app.pipecd.yaml.
+	for i := range manifests {
+		m := manifests[i]
+		err := l.refineNamespace(&m)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return
 }
 
-func setNamespace(manifests []Manifest, namespace string) {
-	if namespace == "" {
-		return
+// refineNamespace fix the namespace of the given manifest.
+// The priority is as follows:
+// If the resource is cluster-scoped, it returns an empty string.
+// Otherwise, it is the namespace-scoped resource and the namespace is determined by the following order:
+// 1. The namespace set in the application configuration.
+// 2. The namespace set in the manifest.
+// 3. The default namespace.
+func (l *loader) refineNamespace(m *Manifest) error {
```

Comment:
Should it be named `determineResourceNamespace` or `determineNamespace`? 🤔

Reply:
Thanks! fixed [5d22c75](https://github.com/pipe-cd/pipecd/pull/4916/commits/5d22c7587fec159fb2a8e926ade63ded2eead06b)